File: feitos/2482 - Etiquetas de Noel/Main.java
import java.util.Scanner;

class Main {
    public static void verificar(Xmas[] x, Pessoa[] p, int n, int m) {
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                if (x[j].lingua.compareTo(p[i].idioma) == 0) {
                    System.out.println(p[i].nome);
                    System.out.println(x[j].msg);
                    j = n;
                }
            }
        }
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();
        sc.nextLine();
        Xmas[] x = new Xmas[n];

        for (int i = 0; i < n; i++) {
            String lingua = sc.nextLine();
            String msg = sc.nextLine();
            x[i] = new Xmas(lingua, msg);
        }

        int m = sc.nextInt();
        sc.nextLine();
        Pessoa[] p = new Pessoa[m];

        for (int i = 0; i < m; i++) {
            String nome = sc.nextLine();
            String idioma = sc.nextLine();
            p[i] = new Pessoa(nome, idioma);
        }
        verificar(x, p, n, m);

        sc.close();
    }
}

class Xmas {
    String lingua;
    String msg;

    public Xmas(String lingua, String msg) {
        this.lingua = lingua;
        this.msg = msg;
    }
}

class Pessoa {
    String nome;
    String idioma;

    public Pessoa(String nome, String idioma) {
        this.nome = nome;
        this.idioma = idioma;
    }
}

File: feitos/2482 - Etiquetas de Noel/main.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

typedef struct {
    char lingua[21];
    char msg[21];
} Xmas;

typedef struct {
    char nome[31];
    char idioma[21];
} Pessoa;

void verificar(Xmas x[], Pessoa p[], int n, int m) {
    for (int i = 0; i < m; i++) {
        for (int j = 0; j < n; j++) {
            if (strcmp(x[j].lingua, p[i].idioma) == 0) {
                printf("%s\n", p[i].nome);
                printf("%s\n", x[j].msg);
                j = n;
            }
        }
    }
}

int main() {
    int n;
    scanf("%d", &n);
    Xmas x[n];

    for (int i = 0; i < n; i++) {
        scanf(" %[^\n]", x[i].lingua);
        scanf(" %[^\n]", x[i].msg);
    }

    int m;
    scanf("%d", &m);
    Pessoa p[m];

    for (int i = 0; i < m; i++) {
        scanf(" %[^\n]", p[i].nome);
        scanf(" %[^\n]", p[i].idioma);
    }
    verificar(x, p, n, m);

    return 0;
}